iree/compiler/InputConversion/MHLO/MHLOToLinalgOnTensors.cpp

```diff
@@ -268,6 +268,101 @@ struct ConstOpConversion : public OpConversionPattern<mhlo::ConstOp> {
   }
 };
 
+//===----------------------------------------------------------------------===//
+// mhlo.RngUniformOp conversion patterns.
+//===----------------------------------------------------------------------===//
+
+namespace {
+/// Returns an ArrayAttr that contains `nLoops` attributes. All the attributes
+/// are "parallel" except the last `nReduction` elements, where are "reduction"
+/// attributes.
+SmallVector<StringRef, 3> GetParallelAndReductionIterators(
+    unsigned nLoops, unsigned nReduction) {
+  SmallVector<StringRef, 3> res(nLoops - nReduction,
+                                getParallelIteratorTypeName());
+  res.append(nReduction, getReductionIteratorTypeName());
+  return res;
+}
+
+struct RngUniformConversion : public OpConversionPattern<mhlo::RngUniformOp> {
+  using OpConversionPattern<mhlo::RngUniformOp>::OpConversionPattern;
+
+  LogicalResult matchAndRewrite(
+      mhlo::RngUniformOp op, ArrayRef<Value> args,
+      ConversionPatternRewriter &rewriter) const final {
+    Type int32Type = IntegerType::get(op.getContext(), /*width=*/32);
+    auto loc = op.getLoc();
+    auto targetTy = op->getResult(0).getType().dyn_cast<ShapedType>();
+    auto targetRank = targetTy.getRank();
+    Value output;
+    Value initTensor = rewriter.create<linalg::InitTensorOp>(
+        loc, ArrayRef<Value>({}), targetTy.getShape(),
+        targetTy.getElementType());
+    auto resultTy = this->typeConverter->convertType(op.getResult().getType())
+                        .cast<ShapedType>();
+    // Creates index map using target matrix's rank.
+    SmallVector<AffineMap, 3> indexingMaps;
+    indexingMaps.push_back(AffineMap::get(targetRank, /*symbolCount=*/0,
+                                          SmallVector<AffineExpr>({}),
+                                          rewriter.getContext()));
+    indexingMaps.push_back(AffineMap::get(targetRank, /*symbolCount=*/0,
+                                          SmallVector<AffineExpr>({}),
+                                          rewriter.getContext()));
```

Comment:
Since we assume both operands are 0-D tensor, this can be

```
SmallVector<AffineMap, 3> indexingMaps(2, AffineMap::get(targetRank, /*symbolCount=*/0,
                                          SmallVector<AffineExpr>({}),
                                          rewriter.getContext()));
```

Reply:
done